utils/common/src/main/kotlin/ProcessCapture.kt

```diff
@@ -70,12 +79,21 @@ class ProcessCapture(
          * Filter out the variables that are not white-listed to be propagated from [environment].
          */
         private fun filterEnvironmentVariables(environment: MutableMap<String, String>): MutableMap<String, String> {
-            val keysToRemove = environment.keys.filterNot { it in variablesToPropagate }
+            val keys = environment.keys.handleCase()
```

Comment:
Alternatively to this, apply `.toSortedMap(String.CASE_INSENSITIVE_ORDER)` to `environment` on Windows before looking up the keys.